File: software/examples/php/ExampleButtonCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletLCD16x2.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletLCD16x2;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your LCD 16x2 Bricklet

// Callback function for button pressed callback
function cb_buttonPressed($button)
{
    echo "Button Pressed: $button\n";
}

// Callback function for button released callback
function cb_buttonReleased($button)
{
    echo "Button Released: $button\n";
}

$ipcon = new IPConnection(); // Create IP connection
$lcd = new BrickletLCD16x2(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register button pressed callback to function cb_buttonPressed
$lcd->registerCallback(BrickletLCD16x2::CALLBACK_BUTTON_PRESSED, 'cb_buttonPressed');

// Register button released callback to function cb_buttonReleased
$lcd->registerCallback(BrickletLCD16x2::CALLBACK_BUTTON_RELEASED, 'cb_buttonReleased');

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: software/examples/c/example_button_callback.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_lcd_16x2.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

// Callback function for button pressed callback
void cb_button_pressed(uint8_t button_pressed, void *user_data) {
	(void)user_data; // avoid unused parameter warning

	printf("Button Pressed: %d\n", button_pressed);
}

// Callback function for button released callback
void cb_button_released(uint8_t button_released, void *user_data) {
	(void)user_data; // avoid unused parameter warning

	printf("Button Released: %d\n", button_released);
}

int main() {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	LCD16x2 lcd162;
	lcd_16x2_create(&lcd162, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		exit(1);
	}
	// Don't use device before ipcon is connected

	// Register button pressed callback to function cb_button_pressed
	lcd_16x2_register_callback(&lcd162,
	                           LCD_16X2_CALLBACK_BUTTON_PRESSED,
	                           (void *)cb_button_pressed,
	                           NULL);

	// Register button released callback to function cb_button_released
	lcd_16x2_register_callback(&lcd162,
	                           LCD_16X2_CALLBACK_BUTTON_RELEASED,
	                           (void *)cb_button_released,
	                           NULL);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
}
